File: unit3/Profile2/app/src/main/java/com/example/profile/MainActivity.java
package com.example.profile;

import android.content.Context;
import android.content.Intent;
import android.content.SharedPreferences;
import android.net.Uri;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.view.View;
import android.widget.TextView;

public class MainActivity extends AppCompatActivity {
    //views
    TextView name;
    TextView role;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        name = findViewById(R.id.profileName);
        role = findViewById(R.id.profileRole);
        //get Data
        SharedPreferences sharedPref = getSharedPreferences("myData", Context.MODE_PRIVATE);
        if (sharedPref.contains("name"))
            name.setText(sharedPref.getString("name", ""));
        if (sharedPref.contains("role"))
            role.setText(sharedPref.getString("role", ""));
    }

    public void editProfile(View view) {
        Intent intent = new Intent(MainActivity.this, EditProfile.class);
        startActivity(intent);
    }

    public void openEmail(View view) {
        //email challenge solution
        Intent emailIntent = new Intent(Intent.ACTION_SEND);
        String[] TO = {"address@email.com"};
        emailIntent.setData(Uri.parse("mailto:"));
        emailIntent.setType("text/plain");
        emailIntent.putExtra(Intent.EXTRA_EMAIL, TO);
        emailIntent.putExtra(Intent.EXTRA_SUBJECT, "Your subject");
        emailIntent.putExtra(Intent.EXTRA_TEXT, "Email message goes here");
        startActivity(Intent.createChooser(emailIntent, "Send mail..."));
    }

    public void openPhone(View view) {
        Uri uri = Uri.parse("tel:0123456789");
        Intent intent = new Intent(Intent.ACTION_DIAL, uri);
        // intent.setData(Uri.parse("tel:0123456789"));
        startActivity(intent);
    }

    public void openTwitter(View view) {
        Intent intent = new Intent(Intent.ACTION_VIEW, Uri.parse("https://twitter.com/Twitter"));
        startActivity(intent);
    }
}

File: Kotlin_Projects/ProfileApp/app/src/main/java/com/example/profileapp/MainActivity.kt
package com.example.profileapp

import android.content.Context
import android.content.Intent
import android.content.SharedPreferences
import android.net.Uri
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import android.widget.TextView

class MainActivity : AppCompatActivity() {
    lateinit var email: TextView
    lateinit var phone: TextView
    lateinit var location: TextView
    lateinit var name: TextView
    lateinit var role: TextView


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        email = findViewById(R.id.email)
        phone = findViewById(R.id.phone)
        location = findViewById(R.id.location)
        name=findViewById(R.id.displayed_name)
        role=findViewById(R.id.displayed_role)

        //get Data
        var shared: SharedPreferences =getSharedPreferences("myData", Context.MODE_PRIVATE)
        if(shared.contains("name")){
            name.text= shared.getString("name","")
        }
        if(shared.contains("role")) {
            role.text = shared.getString("role", "")
        }
    }

    fun openEmail(view: View) {
        val intent = Intent(Intent.ACTION_SENDTO,Uri.parse("mailto: Basma@gmail.com"))
        startActivity(intent)
    }

    fun openPhone(view: View) {
val intent= Intent(Intent.ACTION_DIAL, Uri.parse("tel: 05539864"))
        startActivity(intent)
    }
    fun openLocation(view: View) {
        val intent=Intent(Intent.ACTION_VIEW, Uri.parse("https://twitter.com/Basma_s2"))
        startActivity(intent)
    }

    fun nextPage(view: View) {
        var i = Intent(this,EditProfile::class.java)
        startActivity(i)
    }
}